DebuggerFeaturePkg/Library/DebugAgent/AARCH64/DebugAarch64.c

```diff
@@ -449,3 +462,104 @@ IsPageWritable (
 {
   return CheckPageAccess (Address, TRUE);
 }
+
+/**
+  Adds a AARCH64 hardware watch point.
+
+  @param[in]  Address   The address of the data watch point.
+  @param[in]  Length    The length of the data watch point.
+  @param[in]  Read      Boolean indicated break on read.
+  @param[in]  Write     Boolean indicated break on write.
+
+  @retval  TRUE   The watch point was successfully set.
+  @retval  FALSE  The watch point could not be set.
+**/
+BOOLEAN
+AddWatchpoint (
+  IN UINTN    Address,
+  IN UINTN    Length,
+  IN BOOLEAN  Read,
+  IN BOOLEAN  Write
+  )
+{
+  UINTN    Index;
+  UINTN    Bas;
+  UINTN    Lsc;
+  DBG_WCR  DbgWcr;
+
+  // Byte Address Select is a bitmap where each bit in Address + N up to +7.
+  // shift away full 8 by (8 - count) to get this.
+  Bas = (0xFF >> (8 - MIN (Length, 8)));

```

Comment:
can this be changed to use RShiftU64  from BaseLib?

Reply:
Is this just to avoid sign extension? The same could be done by `0xFFllu`, right?

Reply:
Shift functions can be interpreted by the compiler into an intrinsic call to _allshr/_allshl. The easiest way to prevent this is to use the BaseLib calls to handle shift operations. This has been seen more frequently with IA32 builds.